### ISSUE TITLE
v1.20.0 - Applications now require clean on every build

### ISSUE THREAD

**Issue by khari998** [flutter/flutter#60244]
I upgraded to the latest version 1.20.0-1.0.pre.200. Ever since I have been on v1.20, my application would only build once and then fail every subsequent time. I don't know if this has to do specifically with the iPhone simulator but I am using the iPhone 8 as my simulated device.

I would run the application on the simulator using ```flutter run -d iph``` then after building and compiling successfully, and then closing out of flutter on the command line using ctrl + c, the next time running the command ```flutter run -d iph``` produces this error and the app will fail to build every subsequent time: 

```
Xcode's output:
↳
    building file list ... done
    deleting App.framework/_CodeSignature/CodeResources
    deleting App.framework/_CodeSignature/
    App.framework/
    App.framework/App
    App.framework/flutter_assets/
    App.framework/flutter_assets/AssetManifest.json
    App.framework/flutter_assets/FontManifest.json
    App.framework/flutter_assets/NOTICES
    App.framework/flutter_assets/ios/
    App.framework/flutter_assets/ios/Flutter/
    App.framework/flutter_assets/ios/Flutter/App.framework/
    App.framework/flutter_assets/ios/Flutter/App.framework/flutter_assets/
    App.framework/flutter_assets/ios/Flutter/App.framework/flutter_assets/.en
    v

    sent 775004 bytes  received 166 bytes  1550340.00 bytes/sec
    total size is 39162254  speedup is 50.52
    building file list ... done
    Flutter.framework/
    Flutter.framework/Flutter
    Flutter.framework/Info.plist
    Flutter.framework/icudtl.dat
    Flutter.framework/_CodeSignature/
    Flutter.framework/_CodeSignature/CodeResources

    sent 97430188 bytes  received 120 bytes  21651179.56 bytes/sec
    total size is 97417883  speedup is 1.00
    /Users/Khari/Documents/Dev_Projects/HCD/build/ios/Debug-iphonesimulator/R
    unner.app/Frameworks/App.framework/App: signed bundle with Mach-O
    universal (x86_64 arm64) [io.flutter.flutter.app]
    /Users/Khari/Documents/Dev_Projects/HCD/build/ios/Debug-iphonesimulator/R
    unner.app/Frameworks/Flutter.framework/Flutter: signed bundle with Mach-O
    universal (armv7 x86_64 arm64) [io.flutter.flutter]
    2020-06-24 18:45:21.906 defaults[52508:12016403] 
    The domain/default pair of
    (/Users/Khari/Documents/Dev_Projects/HCD/build/ios/Debug-iphonesimulator/
    Runner.app/Frameworks/App.framework/flutter_assets/ios/Flutter/App.framew
    ork/Info.plist, CFBundleExecutable) does not exist
    fatal error: lipo: can't map input file:
    /Users/Khari/Documents/Dev_Projects/HCD/build/ios/Debug-iphonesimulator/R
    unner.app/Frameworks/App.framework/flutter_assets/ios/Flutter/App.framewo
    rk/ (Invalid argument)
    fatal error: lipo: can't map input file:
    /Users/Khari/Documents/Dev_Projects/HCD/build/ios/Debug-iphonesimulator/R
    unner.app/Frameworks/App.framework/flutter_assets/ios/Flutter/App.framewo
    rk/ (Invalid argument)
    Failed to extract x86_64 for
    /Users/Khari/Documents/Dev_Projects/HCD/build/ios/Debug-iphonesimulator/R
    unner.app/Frameworks/App.framework/flutter_assets/ios/Flutter/App.framewo
    rk/. Running lipo -info:
    fatal error: lipo: can't map input file:
    /Users/Khari/Documents/Dev_Projects/HCD/build/ios/Debug-iphonesimulator/R
    unner.app/Frameworks/App.framework/flutter_assets/ios/Flutter/App.framewo
    rk/ (Invalid argument)
    Command PhaseScriptExecution failed with a nonzero exit code
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description
    warning: Capabilities for Signing & Capabilities may not function
    correctly because its entitlements use a placeholder team ID. To resolve
    this, select a development team in the Runner editor. (in target 'Runner'
    from project 'Runner')
```

The temporary solution is to run ```flutter clean``` and then rebuild using the same command ```flutter run -d iph``` 
After running flutter clean, the app recompiles and launches successfully. But now I have to run ```flutter clean``` every time I exit Flutter in order for the app to successfully compile. 

Flutter doctor -v is a-okay

```
[✓] Flutter (Channel master, 1.20.0-1.0.pre.200, on Mac OS X 10.15.5 19F101,
    locale en-US)
    • Flutter version 1.20.0-1.0.pre.200 at
      /Users/Khari/Documents/Dev_Projects/flutter
    • Framework revision 15913e5d73 (2 hours ago), 2020-06-24 17:13:02 -0400
    • Engine revision 291dab6f17
    • Dart version 2.9.0 (build 2.9.0-18.0.dev d8eb844e5d)

 
[✓] Android toolchain - develop for Android devices (Android SDK version
    29.0.1)
    • Android SDK at /Users/Khari/Library/Android/sdk
    • Platform android-29, build-tools 29.0.1
    • ANDROID_HOME = /Users/Khari/Library/Android/sdk
    • ANDROID_SDK_ROOT = /Users/Khari/Library/Android/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_152-release-1343-b01)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.4, Build version 11E146
    • CocoaPods version 1.8.4

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 3.4)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 37.1.1
    • Dart plugin version 183.6270
    • Java version OpenJDK Runtime Environment (build
      1.8.0_152-release-1343-b01)

[✓] VS Code (version 1.46.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.11.0

[✓] Connected device (3 available)
    • iPhone 8   • A29FA983-71CB-4127-A0AA-C0BA22B84E86 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-13-4 (simulator)
    • Web Server • web-server                           • web-javascript •
      Flutter Tools
    • Chrome     • chrome                               • web-javascript •
      Google Chrome 83.0.4103.116
```

**Comment by Calamity210** [comment]
Can you reproduce with a new project (created with flutter create foo)?

**Comment by khari998** [comment]
I don't know if this is related or not but I am also now failing to build an apk from my app. No idea why this is happening. 

```flutter build apk --split-per-abi```

```
Running Gradle task 'assembleRelease'...                                
                                                                
Running Gradle task 'assembleRelease'...                                                                                                                                FAILURE: Build failed with an exception.
Running Gradle task 'assembleRelease'...                                                                                                                                        
Running Gradle task 'assembleRelease'...                                                                                                                                * What went wrong:
Running Gradle task 'assembleRelease'...                                                                                                                                Could not determine the dependencies of task ':app:compileReleaseJavaWithJavac'.
Running Gradle task 'assembleRelease'...                                                                                                                                > Could not resolve all task dependencies for configuration ':app:releaseCompileClasspath'.
Running Gradle task 'assembleRelease'...                                                                                                                                   > Could not resolve io.flutter:x86_64_release:1.0.0-291dab6f1723db5e7a9ea12109e33702bd224db6.
Running Gradle task 'assembleRelease'...                                                                                                                                     Required by:
Running Gradle task 'assembleRelease'...                                                                                                                                         project :app
Running Gradle task 'assembleRelease'...                                                                                                                                      > Could not resolve io.flutter:x86_64_release:1.0.0-291dab6f1723db5e7a9ea12109e33702bd224db6.
Running Gradle task 'assembleRelease'...                                                                                                                                         > Could not get resource 'https://storage.googleapis.com/download.flutter.io/io/flutter/x86_64_release/1.0.0-291dab6f1723db5e7a9ea12109e33702bd224db6/x86_64_release-1.0.0-291dab6f1723db5e7a9ea12109e33702bd224db6.pom'.
Running Gradle task 'assembleRelease'...                                                                                                                                            > Could not GET 'https://storage.googleapis.com/download.flutter.io/io/flutter/x86_64_release/1.0.0-291dab6f1723db5e7a9ea12109e33702bd224db6/x86_64_release-1.0.0-291dab6f1723db5e7a9ea12109e33702bd224db6.pom'.
Running Gradle task 'assembleRelease'...                                                                                                                                               > Connect to storage.googleapis.com:443 [storage.googleapis.com/2607:f8b0:4004:806:0:0:0:2010] failed: No route to host (connect failed)
Running Gradle task 'assembleRelease'...                                                                                                                                      > Could not resolve io.flutter:x86_64_release:1.0.0-291dab6f1723db5e7a9ea12109e33702bd224db6.
Running Gradle task 'assembleRelease'...                                                                                                                                         > Could not get resource 'https://storage.googleapis.com/download.flutter.io/io/flutter/x86_64_release/1.0.0-291dab6f1723db5e7a9ea12109e33702bd224db6/x86_64_release-1.0.0-291dab6f1723db5e7a9ea12109e33702bd224db6.pom'.
Running Gradle task 'assembleRelease'...                                                                                                                                            > Could not GET 'https://storage.googleapis.com/download.flutter.io/io/flutter/x86_64_release/1.0.0-291dab6f1723db5e7a9ea12109e33702bd224db6/x86_64_release-1.0.0-291dab6f1723db5e7a9ea12109e33702bd224db6.pom'.
Running Gradle task 'assembleRelease'...                                                                                                                                               > Connect to storage.googleapis.com:443 [storage.googleapis.com/2607:f8b0:4004:806:0:0:0:2010] failed: No route to host (connect failed)
Running Gradle task 'assembleRelease'...                                                                                                                                      > Could not resolve io.flutter:x86_64_release:1.0.0-291dab6f1723db5e7a9ea12109e33702bd224db6.
Running Gradle task 'assembleRelease'...                                                                                                                                         > Could not get resource 'https://storage.googleapis.com/download.flutter.io/io/flutter/x86_64_release/1.0.0-291dab6f1723db5e7a9ea12109e33702bd224db6/x86_64_release-1.0.0-291dab6f1723db5e7a9ea12109e33702bd224db6.pom'.
Running Gradle task 'assembleRelease'...                                                                                                                                            > Could not GET 'https://storage.googleapis.com/download.flutter.io/io/flutter/x86_64_release/1.0.0-291dab6f1723db5e7a9ea12109e33702bd224db6/x86_64_release-1.0.0-291dab6f1723db5e7a9ea12109e33702bd224db6.pom'.
Running Gradle task 'assembleRelease'...                                                                                                                                               > Connect to storage.googleapis.com:443 [storage.googleapis.com/2607:f8b0:4004:806:0:0:0:2010] failed: No route to host (connect failed)
Running Gradle task 'assembleRelease'...                                                                                                                                        
Running Gradle task 'assembleRelease'...                                                                                                                                * Try:  
Running Gradle task 'assembleRelease'...                                                                                                                                Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
Running Gradle task 'assembleRelease'...                                                                                                                                        
Running Gradle task 'assembleRelease'...                                                                                                                                * Get more help at https://help.gradle.org
Running Gradle task 'assembleRelease'...                                                                                                                                        
Running Gradle task 'assembleRelease'...                                                                                                                                BUILD FAILED in 2m 18s
Running Gradle task 'assembleRelease'...                                                                                                                           
Running Gradle task 'assembleRelease'... Done                     141.7s (!)
Gradle task assembleRelease failed with exit code 1
```

**Comment by khari998** [comment]
> Can you reproduce with a new project (created with flutter create foo)?

No app I build reproduces the issue.

This causes me to believe it may be related to the fact that I was told my Podfile was out of date and to run the command rm ios/Podfile. I also did this before upgrading to v1.20

After the initial successful build, i tried navigating into the pwd/ios and running pod install and pod update. No dice. 

I also upgraded to cocoapods 1.9.1 after it didn't work.

After upgrading, I removed all pod files from the ios folder and recreated them. 

Still running into this issue.

**Comment by khari998** [comment]
Seems to be related to this https://github.com/flutter/flutter/issues/57210

My app is breaking all the time now and there is no way to fix it so far.

**Comment by khari998** [comment]
I think I am starting to get somewhere.

I had a workaround by downgrading to 1.18 and replacing my ios Folder with a previous version of my ios folder (thank god for commits)

I got my app to run successfully on 1.18 after doing this. 

So of course I try upgrading and I get this error and now my project wont build at a ll on v1.20

```
Xcode's output:
↳
    Error: Could not resolve the package 'characters' in
    'package:characters/characters.dart'.
    ../flutter/packages/flutter/lib/src/material/text_fie
    ld.dart:9:8: Error: Not found:
    'package:characters/characters.dart'
    import 'package:characters/characters.dart';
           ^
    ../flutter/packages/flutter/lib/widgets.dart:18:1:
    Error: Not found:
    'package:characters/characters.dart'
    export 'package:characters/characters.dart';
    ^
    ../flutter/packages/flutter/lib/src/rendering/editabl
    e.dart:10:8: Error: Not found:
    'package:characters/characters.dart'
    import 'package:characters/characters.dart';
           ^
    ../flutter/packages/flutter/lib/src/services/text_for
    matter.dart:9:8: Error: Not found:
    'package:characters/characters.dart'
    import 'package:characters/characters.dart';
           ^
    ../flutter/packages/flutter/lib/src/material/text_fie
    ld.dart:822:61: Error: The getter 'characters' isn't
    defined for the class 'String'.
    Try correcting the name to the name of an existing
    getter, or defining a getter or field named
    'characters'.
      int get _currentLength =>
      _effectiveController.value.text.characters.length;
                                                                ^^^^^^^^^^
    ../flutter/packages/flutter/lib/src/material/text_fie
    ld.dart:824:118: Error: The getter 'characters' isn't
    defined for the class 'String'.
    Try correcting the name to the name of an existing
    getter, or defining a getter or field named
    'characters'.
      bool get _hasIntrinsicError => widget.maxLength !=
      null && widget.maxLength > 0 &&
      _effectiveController.value.text.characters.length >
      widget.maxLength;
                                                                                                                         ^^^^^^^^^^
    ../flutter/packages/flutter/lib/src/rendering/editabl
    e.dart:244:64: Error: The getter 'characters' isn't
    defined for the class 'String'.
    Try correcting the name to the name of an existing
    getter, or defining a getter or field named
    'characters'.
           assert(obscuringCharacter != null &&
           obscuringCharacter.characters.length == 1),
                                                                   ^^^^^^^^^^
    ../flutter/packages/flutter/lib/src/rendering/editabl
    e.dart:361:35: Error: The getter 'characters' isn't
    defined for the class 'String'.
    Try correcting the name to the name of an existing
    getter, or defining a getter or field named
    'characters'.
        assert(value != null && value.characters.length
        == 1);
                                      ^^^^^^^^^^
    ../flutter/packages/flutter/lib/src/rendering/editabl
    e.dart:567:11: Error: 'Characters' isn't a type.
        final Characters remaining =
        string.characters.skipWhile((String
        currentString) {
              ^^^^^^^^^^
    ../flutter/packages/flutter/lib/src/rendering/editabl
    e.dart:567:41: Error: The getter 'characters' isn't
    defined for the class 'String'.
    Try correcting the name to the name of an existing
    getter, or defining a getter or field named
    'characters'.
        final Characters remaining =
        string.characters.skipWhile((String
        currentString) {
                                            ^^^^^^^^^^
    ../flutter/packages/flutter/lib/src/rendering/editabl
    e.dart:600:47: Error: The getter 'characters' isn't
    defined for the class 'String'.
    Try correcting the name to the name of an existing
    getter, or defining a getter or field named
    'characters'.
        for (final String currentString in
        string.characters) {
                                                  ^^^^^^^^^^
    ../flutter/packages/flutter/lib/src/rendering/editabl
    e.dart:602:40: Error: The getter 'characters' isn't
    defined for the class 'String'.
    Try correcting the name to the name of an existing
    getter, or defining a getter or field named
    'characters'.
              !_isWhitespace(currentString.characters.fir
              st.toString().codeUnitAt(0))) {
                                           ^^^^^^^^^^
    ../flutter/packages/flutter/lib/src/services/text_for
    matter.dart:355:11: Error: 'CharacterRange' isn't a
    type.
        final CharacterRange iterator =
        CharacterRange(value.text);
              ^^^^^^^^^^^^^^
    ../flutter/packages/flutter/lib/src/services/text_for
    matter.dart:355:37: Error: Method not found:
    'CharacterRange'.
        final CharacterRange iterator =
        CharacterRange(value.text);
                                        ^^^^^^^^^^^^^^
    ../flutter/packages/flutter/lib/src/services/text_for
    matter.dart:356:20: Error: The getter 'characters'
    isn't defined for the class 'String'.
    Try correcting the name to the name of an existing
    getter, or defining a getter or field named
    'characters'.
        if (value.text.characters.length > maxLength) {
                       ^^^^^^^^^^
    ../flutter/packages/flutter/lib/src/services/text_for
    matter.dart:375:61: Error: The getter 'characters'
    isn't defined for the class 'String'.
    Try correcting the name to the name of an existing
    getter, or defining a getter or field named
    'characters'.
        if (maxLength != null && maxLength > 0 &&
        newValue.text.characters.length > maxLength) {
                                                                ^^^^^^^^^^
    ../flutter/packages/flutter/lib/src/services/text_for
    matter.dart:378:25: Error: The getter 'characters'
    isn't defined for the class 'String'.
    Try correcting the name to the name of an existing
    getter, or defining a getter or field named
    'characters'.
          if (oldValue.text.characters.length ==
          maxLength) {
                            ^^^^^^^^^^
    Unhandled exception:
    FileSystemException(uri=org-dartlang-untranslatable-u
    ri:package%3Acharacters%2Fcharacters.dart;
    message=StandardFileSystem only supports file:* and
    data:* URIs)
    #0      StandardFileSystem.entityForUri
    (package:front_end/src/api_prototype/standard_file_sy
    stem.dart:33:7)
    #1      asFileUri
    (package:vm/kernel_front_end.dart:659:37)
    #2      writeDepfile
    (package:vm/kernel_front_end.dart:853:21)
    <asynchronous suspension>
    #3      FrontendCompiler.compile
    (package:frontend_server/frontend_server.dart:574:15)
    <asynchronous suspension>
    #4      _FlutterFrontendCompiler.compile
    (package:flutter_frontend_server/server.dart:42:22)
    #5      starter
    (package:flutter_frontend_server/server.dart:181:27)
    #6      main
    (file:///b/s/w/ir/cache/builder/src/flutter/flutter_f
    rontend_server/bin/starter.dart:8:30)
    #7      _startIsolate.<anonymous closure>
    (dart:isolate-patch/isolate_patch.dart:299:32)
    #8      _RawReceivePortImpl._handleMessage
    (dart:isolate-patch/isolate_patch.dart:168:12)


    Command PhaseScriptExecution failed with a nonzero
    exit code
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description
```

I am also getting a warning saying my Podfile is out of date. But I am not going to run the command ```rm ios/Podfile```` this time because everything started breaking after I ran that command the last time.

```
Warning: Podfile is out of date
  This can cause issues if your application depends on
  plugins that do not support iOS.
  See
  https://flutter.dev/docs/development/packages-and-plugi
  ns/developing-packages#plugin-platforms for details.
  If you have local Podfile edits you would like to keep,
  see https://github.com/flutter/flutter/issues/45197 for
  instructions.
To regenerate the Podfile, run:
  rm ios/Podfile
```

I hope that error message is enough to help you guys narrow down the issue as it appears to be coming from some line in the framework after upgrading. For now I will downgrade to 1.18 until the cause of this can be properly identified.

**Comment by khari998** [comment]
After moving back to 1.18, I no longer get a message about my Podfile being out of date and everything compiles like a charm. I really hope the root of this problem can be solved so I can stay on the most recent versions of Flutter.

**Comment by jmagman** [comment]
Thank you for the thorough investigation!
Would you mind attaching these files in your project root?  You don't need to re-upgrade, they should be the same on 1.18.

pubspec.yaml
.flutter-plugins
.flutter-plugins-dependencies

**Comment by jmagman** [comment]
@khari998 
1. Actually, if you can reproduce the original error on 1.20 with the -v verbose flag, that would be really helpful: `flutter run -d iph -v`

2. When you're reproducing, does this file exist?
/Users/Khari/Documents/Dev_Projects/HCD/build/ios/Debug-iphonesimulator/Runner.app/Frameworks/App.framework/App
3. 
```
    ld.dart:9:8: Error: Not found:
    'package:characters/characters.dart'
```
cache is messed up between upgrade and downgrade?  Does `flutter update-packages` fix that one?

**Comment by jmagman** [comment]
> cache is messed up between upgrade and downgrade? Does flutter update-packages fix that one?

Never mind, try `flutter pub get` for this one, not `flutter update-packages`

**Comment by jmagman** [comment]
Duplicate of https://github.com/flutter/flutter/issues/57210.

**Comment by khari998** [comment]
> 2\. When you're reproducing, does this file exist?

This file does not exist for some reason. It happened after updating my Podfile. 

I havent gone back to 1.20 yet. I am on 1.18 because I need to keep my app working for the client I am building it for.

Flutter pub get did not work when I tried it before

**Comment by khari998** [comment]
Update, 1.18 does not work and I still have the same issue

**Comment by TahaTesser** [comment]
Please check this https://github.com/flutter/flutter/issues/60244#issuecomment-649334173
Closing as duplicate of https://github.com/flutter/flutter/issues/57210
